org.eclipse.xtext.xbase.ui/xtend-gen/org/eclipse/xtext/xbase/ui/editor/XbaseEditorInputRedirector.java

```diff
@@ -50,7 +50,7 @@
   public ITypeRoot getTypeRoot(final IEditorInput it) {
     ITypeRoot _xblockexpression = null;
     {
-      final Object adapter = it.getAdapter(IJavaElement.class);
+      final IJavaElement adapter = it.<IJavaElement>getAdapter(IJavaElement.class);
```

Comment:
be careful: this will make this Java source file compilable (and runnable) only when using the Neon target platform: since adapters are type paremeterized only starting from Neon.

I think we discussed this problem in another PR or issue